### ISSUE TITLE
Bare ampersands are not detected in attributes

### ISSUE THREAD

**Issue by ksmadsen** [petdance/html-lint#21]
Hi,

If a bare ampersand occurs within an attribute (e.g. a href), it is not detected as being invalid. Likewise for unknown entities.

Testcase:

``` perl
use Test::HTML::Lint tests => 2;
html_ok('<a href="&">test</a>');
html_ok('<a href="&noentityisnamedlikethis;">test</a>');
```

The above program produces 2 OK results, even though the two entities are unclosed and unknown.

We are using HTML::Lint version 2.20 on Perl 5.14.2.
